domain/ide/workspace/workspace.go

```diff
@@ -62,10 +62,21 @@ func Set(w *Workspace) {
 	instance = w
 }
 
-func (w *Workspace) DeleteFolder(folder string) {
+func (w *Workspace) DeleteFolder(folderPath string) {
 	w.mutex.Lock()
 	defer w.mutex.Unlock()
-	delete(w.folders, folder)
+	folder := w.GetFolderContaining(folderPath)
+	folder.ClearDiagnosticsFromPathRecursively(folderPath)
```

Comment:
How about we do null-check folder here, similarly to the method below?

Reply:
Done